lib/client/mcp/platform/platform.go

```diff
@@ -0,0 +1,125 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package platform
+
+import (
+	"context"
+	"io"
+	"log/slog"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/types"
+	clientmcp "github.com/gravitational/teleport/lib/client/mcp"
+
+	"github.com/gravitational/trace"
+	mcpserver "github.com/mark3labs/mcp-go/server"
+)
+
+// PlatformAPIClient defines the API client used by platform MCP servers.
+type PlatformAPIClient interface {
+	// GetAccessRequests retrieves a list of all access requests matching the provided filter.
```

Comment:
nit: reuse interface defined in `lib/services`? you can refactor existing ones if needed